From 10e1ad00b861dad49039cea00d4b6a2057b2739f Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 16 Apr 2013 13:36:00 +0900
Subject: [PATCH 02/17] ARM: shmobile: bockw: enable ICB clock

No device drivers enables the ICB clock, so this patch enables
the clock at the board initialization.
---
 arch/arm/mach-shmobile/board-bockw.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-shmobile/board-bockw.c b/arch/arm/mach-shmobile/board-bockw.c
index 059a622..e063de7 100644
--- a/arch/arm/mach-shmobile/board-bockw.c
+++ b/arch/arm/mach-shmobile/board-bockw.c
@@ -415,6 +415,7 @@ static struct i2c_board_info i2c0_devices[] = {
 static void __init bockw_init(void)
 {
 	u16 val;
+	struct clk *icb;
 
 	/*
 	 * CAUTION
@@ -488,6 +489,11 @@ static void __init bockw_init(void)
 
 	r8a7778_add_standard_devices();
 	platform_add_devices(bockw_devices, ARRAY_SIZE(bockw_devices));
+
+	/* enable ICB clock */
+	icb = clk_get(NULL, "icb");
+	if (!IS_ERR(icb))
+		clk_enable(icb);
 }
 
 MACHINE_START(BOCKW, "bockw")
-- 
1.7.9.5

